### PR TITLE
Treat warnings as errors in CI

### PR DESCRIPTION
**Pull request by jturner314**
This helps avoid accidentally missing a warning.

### CODE REVIEW

**Comment by bluss** [comment]
I'm thinking we maybe want to have the continue on error flag set on the clippy job (so that other tests also run) (?)

File: tests/iterators.rs [hunk 1]
```diff
@@ -544,9 +543,9 @@ fn axis_chunks_iter_corner_cases() {
     assert_equal(
         it,
         vec![
-            arr2(&[[7.], [6.], [5.]]),
-            arr2(&[[4.], [3.], [2.]]),
-            arr2(&[[1.], [0.]]),
```
Comment (bluss) [inline tests/iterators.rs:549]: what was the warning here?

Comment (jturner314) [reply tests/iterators.rs:549]: Unused import of `arr2` when `std` feature is not enabled.

**Comment by jturner314** [comment]
> I'm thinking we maybe want to have the continue on error flag set on the clippy job (so that other tests also run) (?)

I think they were running. (I was just force-pushing to fix warnings before they got a chance to finish.) I have seen items canceled when other items fail, but from what I've seen, it's always been within a job (i.e. within "tests" or within "cross_test"), so the tests should still run if clippy fails. I'll check to make sure.

**Comment by bluss** [comment]
I think you'd want continue on error, but yeah I wonder why they are running anyway.

https://github.com/petgraph/petgraph/blob/master/.github/workflows/ci.yml#L45

**Comment by jturner314** [comment]
I came across these pages regarding continue-on-error:

- https://stackoverflow.com/q/62045967
- https://github.community/t/continue-on-error-allow-failure-ui-indication/16773

which seem to indicate that continue-on-error makes the job always "pass", i.e. get a green check. They suggest to use `if: succeeded() || failed()` in the `run` step if you want a job to continue running after a step fails or `if: always()` if you want it to also continue running after a step is canceled.

I'll try adding `continue-on-error` to the (currently failing) clippy job to see what happens.

**Comment by bluss** [comment]
Thanks

**Pull request merged**